### PR TITLE
added controller protobuf

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: protobuf/common/error.proto [hunk 1]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package common;
+
+option go_package = "github.com/layer5io/meshkit/protobuf/common;common";
+
+message CommonError {
```
Comment (mgfeller) [inline protobuf/common/error.proto:7]: Just `Error`? It's `common` as it is in a package called `common`, isn't it?
Will there be other types of errors, i.e. not `Common`, here in the `common` package?

File: protobuf/controller/health.proto [hunk 2]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package controller;
+
+option go_package = "github.com/layer5io/meshkit/protobuf/controller;controller";
+
+enum ControllerStatus {
```
Comment (mgfeller) [inline protobuf/controller/health.proto:7]: Is it common to prepend entities with package name?

File: protobuf/controller/health.proto [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	RUNNING = 0;
+	IDLE = 1;
```
Comment (mgfeller) [inline protobuf/controller/health.proto:9]: unfamiliar with running vs idle - never came across idle as status in software....

**Review by mgfeller — COMMENTED** [review]
@kumarabd could you please point me to some design doc or similar about the controller?

**Comment by kumarabd** [comment]
> @kumarabd could you please point me to some design doc or similar about the controller?

Sure yes @mgfeller - https://docs.google.com/document/d/1AMi5UTUwLRFeQRQj7YuRQgMG-4IXWie24z4R88Eoazs/edit

File: protobuf/common/error.proto [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    string code = 1;
+    string severity = 2;
+    string shortDescription = 3;
```
[the remark this answers is not in the document]
Comment (kumarabd) [reply protobuf/common/error.proto:10]: @leecalcote Same question what I had im mind

File: protobuf/common/error.proto [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    string longDescription = 4;
```
[the remark this answers is not in the document]
Comment (kumarabd) [reply protobuf/common/error.proto:11]: @leecalcote Same question what I had im mind

File: protobuf/common/error.proto (hunk 1 above)
Comment (kumarabd) [reply protobuf/common/error.proto:7]: The redundancy is mandatory here as if we use just `Error` in common package then you cannot use `Error` in any package that imports this package. @mgfeller

File: protobuf/controller/health.proto (hunk 2 above)
Comment (kumarabd) [reply protobuf/controller/health.proto:7]: Yes same as explained above, if we use `Status` here then any package that imports this one cannot have `Status` defined anywhere.

File: protobuf/controller/health.proto (hunk 3 above)
Comment (kumarabd) [reply protobuf/controller/health.proto:9]: So the use case is to get the status of the pipeline defined in the controller, @mgfeller you can suggest any better language here that you think would be good to use :)

File: protobuf/common/error.proto (hunk 4 above)
[the remark this answers is not in the document]
Comment (kumarabd) [reply protobuf/common/error.proto:10]: Pulling this in discussion to the doc and going ahead with the existing way.

**Pull request merged**